R/shiny_mapper.R

```diff
@@ -3,46 +3,51 @@
 #' @param raster_data Raster data in the form or a RasterStack or RasterBrick
 #' @param polygon_data Polygon data to plot on map
 #' @param date_format A date format that will be passed in to lubridate::as_date
+#' @param palette Colour palette to use
+#'
+#' @importFrom magrittr %>%
 #'
 #' @return shinyApp
 #' @keywords internal
-raster_mapping_app <- function(raster_data = NULL, polygon_data = NULL, date_format = NULL) {
-    # TODO - can we check the projection of the data?
-    # TODO - check projection of the polygon data? Or require users to do this?
-    # Remove any letters from the names of the RasterLayers
-    cleaned_names <- stringr::str_replace_all(names(raster_data), pattern = "[A-Za-z]", replacement = "")
-    # Create Date objects
-    date_strings <- lubridate::as_date(cleaned_names, format = date_format)
-    date_strings <- base::sort(date_strings)
-
-    # if (any(is.na(lubridate::parse_date_time(names(raster_data), orders = date_format)))) {
-    #     stop(paste(
-    #         "Unable to parse dates from layer names. Please ensure they are named correctly",
-    #         "or pass the date_format argument with the correct format."
-    #     ))
-    # }
-
-    # We want to be able to lookup the RasterLayer names using the date so we create a list
-    date_list <- as.list(base::sort(names(raster_data)))
-    # Then we assign the date strings we'll use in the UI as the name (or key) for each element in the list
-    names(date_list) <- date_strings
-
-    # Get the extent of the RasterLayer in lat/long
-    raster_extents <- raster::extent(raster::projectRaster(raster_data, crs = "+proj=longlat"))
-    lat_min <- raster_extents@ymin
-    lat_max <- raster_extents@ymax
-    long_min <- raster_extents@xmin
-    long_max <- raster_extents@xmax
-
-    valid_layers <- list("raster")
-    if (!is.null(polygon_data)) valid_layers <- append("polygon", valid_layers)
-
-
-
-    # Define UI for application that draws a histogram
-    ui <- shiny::fillPage(
-        shiny::tags$head(
-            shiny::tags$style(shiny::HTML("
+raster_mapping_app <- function(raster_data = NULL, polygon_data = NULL, date_format = NULL, palette = NULL) {
+  # TODO - can we check the projection of the data?
+  # TODO - check projection of the polygon data? Or require users to do this?
+
+  # Remove any letters from the names of the RasterLayers
+  cleaned_names <- stringr::str_replace_all(names(raster_data), pattern = "[A-Za-z]", replacement = "")
```

Comment:
Is stringr in the Description file?

Reply:
I've just added it, thanks for catching that.